profiler/upload_test.go

```diff
@@ -159,15 +186,28 @@ func BenchmarkDoRequest(b *testing.B) {
 	}
 }
 
-type testServerWaiter func() (http.Header, map[string]string, []string)
+type (
+	// testServerWaiter blocking, captures requests sent to test server for verification
+	testServerWaiter func() (http.Header, map[string]string, []string)
 
-func makeTestServer(t *testing.T, statusCode int) (*httptest.Server, *url.URL, testServerWaiter) {
+	// makeServerFn creates test server given an http.Handler
+	makeServerFn func(handler http.Handler) (srv io.Closer, address string)
+
+	// startTestServerFn starts a test server, returns a server closer, address, and function that
+	// blocks until test server receives a request and captures request data for testing.
+	// Implementations: startHTTPTestServer, startSocketTestServer
+	startTestServerFn func(*testing.T, int) (srv io.Closer, address string, requestWaiter testServerWaiter)
```

Comment:
Let's get rid of this type. It really has no purpose outside of the test where it is used. Alternatively, move the `type` declaration with the documentation into the test and out of the global scope please.

Reply:
moot